packages/flutter_tools/lib/src/flutter_manifest.dart

```diff
@@ -485,6 +529,33 @@ void _validateFlutter(YamlMap yaml, List<String> errors) {
         break;
       case 'generate':
         break;
+      case 'deferred-components':
```

Comment:
This validation is substantially more complex than the other fields, could you pull it into a helper method?

Reply:
Also _validateListType might accomplish some of what you're doing here